check_puppetdb_node.rb

```diff
@@ -10,24 +10,48 @@
 require 'time'
 require 'date'
 
-def api_call(protocol="http", host="localhost", port="8080", query)
+def api_call(query, opt = {})
+  protocol     = opt.has_key?(:ssl) ? 'https' : 'http'
+  host         = opt[:host] || 'localhost'
+  port         = opt[:port] || opt.has_key?(:ssl) ? 8081 : 8080
   uri          = URI("#{protocol}://#{host}:#{port}/#{query}")
   http         = Net::HTTP.new(uri.host, uri.port)
-  http.use_ssl = true  if protocol == 'https'
-  http.use_ssl = false if protocol == 'http'
-  request      = Net::HTTP::Get.new(uri.request_uri, {'Accept' => 'application/json'})
-  response     = http.request(request)
+  http.use_ssl = opt.has_key?(:ssl)
+  if http.use_ssl?
+    require 'openssl/x509'
+    require 'openssl/pkey'
+
+    http.ca_file = opt[:ca] if opt.has_key?(:ca)
+    if opt.has_key?(:cert) and opt.has_key?(:key)
+      http.cert = OpenSSL::X509::Certificate.new(File.read(opt[:cert]))
+      http.key  = OpenSSL::PKey::RSA.new(File.read(opt[:key]))
```

Comment:
Adds support for not only RSA private keys:
```suggestion
      http.key  = OpenSSL::PKey.read(File.read(opt[:key]))
```

Reply:
Tested and added.